devito/passes/iet/misc.py

```diff
@@ -204,3 +207,39 @@ def remove_redundant_moddims(iet):
     iet = Transformer(subs, nested=True).visit(iet)
 
     return iet
+
+
+def abridge_dim_names(iet):
+    subs = {}
+    mapper = {}
+    tree = retrieve_iteration_tree(iet)
+    for it in tree:
```

Comment:
also, tiny comment -- what is this block supposed to do?

Reply:
or perhaps instead of "what", "why" ?

Reply:
It gets all the indices which are `SubDimension`s or `SubDimension`-derived appearing in the expression in the innermost loop nests to assemble substitutions for them. It then checks if it can rename these dimensions to the name of their root without introducing a clash within the inner loop.

However, due to loop structures like
```
for x in range(x_M):
    for yi in range(yi_M):
        ...
    for yj in range(yj_M):
        ....
```
you need to create a global mapper rather than building and applying it on a tree-by-tree basis. Since the mapper isn't complete until the end of the loop, you need to finish the loop before moving onto updating the parents of the unbound indices.

Reply:
Actually, nvm, seems to have been left over from previous iterations. Has now been condensed down